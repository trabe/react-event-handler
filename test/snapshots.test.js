import React from "react";
import { shallow } from "enzyme";
import EventHandler from "../src/event-handler";

const buildHandlerName = eventName => "on" + eventName.charAt(0).toUpperCase() + eventName.slice(1);

const testEvent = (eventName, delay) =>
  done => {
    const start = Date.now();

    const handler = () => {
      try {
        expect(Date.now() - start).toBeGreaterThan(delay || 0);
        done();
      } catch (err) {
        done.fail();
      }
    };

    const wrapper = shallow(
      <EventHandler {...{ [buildHandlerName(eventName)]: delay ? { handler, delay } : handler }}>
        <div>hey!</div>
      </EventHandler>,
    );

    wrapper.find("div").simulate(eventName);
  };

describe("Snapshots", () => {
  describe("EventHandler", () => {
    it("should render its child", () => {
      const wrapper = shallow(
        <EventHandler>
          <div>this should be seen</div>
        </EventHandler>,
      );
      expect(wrapper).toMatchSnapshot();
    });

    ["click", "mouseEnter", "mouseLeave", "focus", "blur"].forEach(eventName => {
      it(`should invoke ${buildHandlerName(eventName)} on child's ${eventName}`, testEvent(eventName));
      it(`should invoke ${buildHandlerName(eventName)} on child's ${eventName} after 50ms`, testEvent(eventName, 50));
    });
  });
});
