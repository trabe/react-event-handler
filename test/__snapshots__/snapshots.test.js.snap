// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshots EventHandler should render its child 1`] = `
<div>
  this should be seen
</div>
`;
